File: 1.0/src/cz/mg/vulkan/PFNvkDestroyPipelineLayout.java
package cz.mg.vulkan;

public class PFNvkDestroyPipelineLayout extends VkFunctionPointer {
    public PFNvkDestroyPipelineLayout() {
    }

    protected PFNvkDestroyPipelineLayout(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkDestroyPipelineLayout(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkDestroyPipelineLayout(long value) {
        setValue(value);
    }

    public PFNvkDestroyPipelineLayout(VkInstance instance) {
        super(instance, new VkString("vkDestroyPipelineLayout"));
    }

    public void call(VkDevice device, VkPipelineLayout pipelineLayout, VkAllocationCallbacks pAllocator){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pipelineLayout != null ? pipelineLayout.getVkAddress() : VkPointer.getNullAddressNative(), pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long device, long pipelineLayout, long pAllocator);



}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkDestroyPipelineLayout.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkDestroyPipelineLayout_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pipelineLayout, jlong pAllocator){
    (void)env;
    (void)jc;
    PFN_vkDestroyPipelineLayout f = (PFN_vkDestroyPipelineLayout)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkPipelineLayout*)jniLongToPointer(pipelineLayout)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator))
    );
}

